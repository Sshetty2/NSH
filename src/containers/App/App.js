import React, { Fragment, useEffect } from 'react';
import SearchBar from '../SearchBar/SearchBar';
import Filters from '../Filters/Filters';
import EventContainer from '../EventContainer/EventContainer';
import Profile from '../Profile/Profile'
import NavBar from '../../components/NavBar/NavBar';
import { withRouter, Route } from 'react-router-dom';
import { connect } from 'react-redux';

export const App = (props) => {

  useEffect(() => {
    
  })

  return (
    <div className="App">
      <NavBar />
      <Route exact path='/' render={() => 
        <Fragment>
          <SearchBar />
          <Filters />
          <EventContainer />
        </Fragment>
      } />
      <Route path='/profile' component={Profile} />
    </div>
  );
}

const mapDispatchToProps = ({

})

export default withRouter(connect(null, mapDispatchToProps)(App));