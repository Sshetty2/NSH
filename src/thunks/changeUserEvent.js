import { setUserEvents, setLoading, setError } from '../actions';
import { getUserEvents } from './getUserEvents';
import { fetchData } from '../utils/api';

export const changeUserEvent = (eventId, method, status = '') => {
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      const path = status.length ? `?status=${status}` : status;
      await fetchData(`/users/1/events/${eventId}` + path, method);
      const events = await getUserEvents('1');
      dispatch(setUserEvents(events.data));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false))
  }
}