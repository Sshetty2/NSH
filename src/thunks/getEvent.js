import { setLoading, setError } from '../actions';
import { fetchData } from '../utils/api';

export const getEvent = (eventId = '') => {
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      const event = await fetchData(`/events/${eventId}`, 'GET');
      dispatch(setLoading(false));
      return event.data.attributes;
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  }
}