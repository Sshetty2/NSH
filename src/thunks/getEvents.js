import { setEvents, setLoading, setError } from '../actions';
import { fetchData } from '../utils/api';

export const getEvents = (limit = null, event_type = null, state = null, month = null) => {
  return async (dispatch) => {
    try {
      dispatch(setLoading(true));
      let path = (limit || event_type || state || month) ? '?' : '';
      if (limit) path = path + `&limit=${limit}`;
      if (event_type) path = path + `&event-type=${event_type}`;
      if (state) path = path + `&state=${state}`;
      if (month) path = path + `&month=${month}`;
      const events = await fetchData('/events' + path, 'GET');
      dispatch(setEvents(events.data));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false))
  }
}