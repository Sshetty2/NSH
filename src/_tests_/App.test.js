import React, { useEffect } from 'react';
import { App, mapStateToProps, mapDispatchToProps } from '../containers/App/App';
import { shallow } from 'enzyme';
import { getUser } from '../thunks/getUser';
import { getEvents } from '../thunks/getEvents';
jest.mock('../thunks/getUser');
jest.mock('../thunks/getEvents');

describe('App', () => {
  let wrapper;
  let mockError;
  let mockLocation;

  beforeEach(() => {
    mockError = '';
    mockLocation = { pathname: '/' };

    wrapper = shallow(
      <App
        getEvents={getEvents}
        getUser={getUser}
        error={mockError}
        location={mockLocation}
      />
    )
  });

  it('should match the correct snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should return an object of props', () => {
      const mockState = {
        error: 'Error message',
        otherState: false,
      };
      const expected = {
        error: 'Error message',
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch when getEvents is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = getEvents();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.getEvents();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('should call dispatch when getUser is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = getUser();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.getUser();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});