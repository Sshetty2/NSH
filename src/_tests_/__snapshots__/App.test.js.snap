// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the correct snapshot 1`] = `
<div
  className="App"
>
  <NavBar />
  <Switch>
    <Route
      component={[Function]}
      path="/profile"
    />
    <Route
      path="/"
      render={[Function]}
    />
  </Switch>
  <Switch>
    <Route
      component={[Function]}
      path="/event/:id"
    />
    <Route
      component={[Function]}
      path="/profile/event/:id"
    />
  </Switch>
</div>
`;
